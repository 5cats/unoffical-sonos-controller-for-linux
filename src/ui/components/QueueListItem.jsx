import port from '../port';
import model from '../model';

import AlbumArt from './AlbumArt';

import React from 'react/addons';

class QueueListItem {

	render () {

		var track = this.props.data;

		return (
			<li onDoubleClick={this._onDoubleClick} data-position={this.props.position}>
				<AlbumArt id="" src={track.albumArtURI} />
				<div className="trackinfo">
					<p className="title">{track.title}</p>
					<p className="artist">{track.creator}</p>
				</div>
			</li>
		);
	}

	_onDoubleClick () {

		port.postMessage({
			type: 'goto',
			host: model.coordinator.host,
			target: this.props.position,
		});

	}
}

QueueListItem.prototype.displayName = "QueueListItem";
export default React.createClass(QueueListItem.prototype);