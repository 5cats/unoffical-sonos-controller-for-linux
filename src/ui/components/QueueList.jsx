import model from '../model';
import register from '../helpers/registerComponent';

import QueueListItem from './QueueListItem';

class QueueList {

	getInitialState () {
		return {
			items: []
		};
	}

	componentDidMount () {
		var self = this;

		model.observe('queue', function () {
			self.setState({
				items: model.queue.items
			});
		});
	}


	render () {

		var queueItemNodes = this.state.items.map(function (i) {
			return (
				<QueueListItem data={i} />
			);
		});

		return (
			<ul id="queue-container">
				{{queueItemNodes}}
			</ul>
		);
	}
}

QueueList.prototype.displayName = "QueueList";
export default register(QueueList);