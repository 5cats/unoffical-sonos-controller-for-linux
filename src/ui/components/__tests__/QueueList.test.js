import { h } from 'preact';
import { QueueList } from '../QueueList';
import QueueListItem from '../QueueListItem';
import { deep } from 'preact-render-spy';

jest.mock('../QueueListItem');
QueueListItem.mockReturnValue(<p className="queue-list-item" />);

describe('QueueList', () => {
    it('renders and matches snapshot', () => {
        const props = {
            tracks: [
                {
                    id: 'foo'
                }
            ]
        };
        const context = deep(<QueueList {...props} />);
        expect(context.output()).toMatchSnapshot();
    });

    it('clicking executed props.flush clickHandler', () => {
        const flush = jest.fn();

        const props = {
            tracks: [
                {
                    id: 'foo'
                }
            ],
            flush
        };
        const context = deep(<QueueList {...props} />);
        context.find('#queue-clear-button').simulate('click');
        expect(flush.mock.calls.length).toBe(1);
    });
});
