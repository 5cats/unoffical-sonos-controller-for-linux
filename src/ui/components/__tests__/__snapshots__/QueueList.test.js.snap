// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueueList renders and matches snapshot 1`] = `
<div
  class="collapsed"
  id="queue-list-container-wrapper"
>
  <h4
    id="queue"
  >
    QUEUE
  </h4>
  <div
    id="queue-list-container"
  >
    <a
      id="queue-clear-button"
      title="Clear Queue"
    >
      <i
        class="material-icons"
      >
        clear_all
      </i>
    </a>
    <ul
      id="queue-container"
    >
      <div
        class="scrollcontainer"
      >
        <p
          class="queue-list-item"
        />
      </div>
    </ul>
  </div>
</div>
`;
